### ISSUE TITLE
RolloverRequestBuilder mapping method

### ISSUE THREAD

**Issue by farchanjo** [elastic/elasticsearch#37400]
```java
public RolloverRequestBuilder mapping(String type, String source) {
     this.request.getCreateIndexRequest().mapping(type, source);
     return this;
}
```
Should be:
```java
public RolloverRequestBuilder mapping(String type, String... source) {
     this.request.getCreateIndexRequest().mapping(type, source);
     return this;
}
```
Because I'm trying to create a mapping on each Rollover new index, but I'm getting:

>mapping source must be pairs of fieldnames and properties definition.

Looking the API I discovered the following issues:
RolloverRequestBuilder.mapping calls:
```java
    public RolloverRequestBuilder mapping(String type, String source) {
        this.request.getCreateIndexRequest().mapping(type, source);
        return this;
    }
```
but getCreateIndexRequest().mapping has a check:

```java
    public CreateIndexRequest mapping(String type, Object... source) {
        mapping(type, PutMappingRequest.buildFromSimplifiedDef(type, source));
        return this;
    }
```
and  XContentBuilder.buildFromSimplifiedDef validate if source is an even value.
```java
    public static XContentBuilder buildFromSimplifiedDef(String type, Object... source) {
        if (source.length % 2 != 0) {
            throw new IllegalArgumentException("mapping source must be pairs of fieldnames and properties definition.");
        }
```
if RolloverRequestBuilder mapping(String type, String source) receive only one parameter as a source string, I will always get an error.  is that right? Or I'm getting crazy. I have tried to add my mapping on rolling index and I'm always getting an error.

Thanks,
Fabricio

**Comment by farchanjo** [comment]
If I'm right, can I submit a fix? 

Thanks,
Fabricio

**Comment by dakrone** [comment]
Hi @farchanjo I took a look at what you pointed out, it does look like it should be changed to accept the multi-arity version of `String... source`.

> If I'm right, can I submit a fix?

Yes please do! Feel free to ping me on the pull request when you open it.

**Comment by farchanjo** [comment]
@dakrone 
I changed it to use Object... like the CreateIndexRequest. 

Can you look my merge, pls? 
https://github.com/elastic/elasticsearch/pull/37401

I added the  RolloverRequestBuilder mapping(String type, String source, XContentType xContentType) signature, because make easier to add new mappings.

I attached an example to the pull request. 

I have build a new artifact and using at my server now and is working. I've made all integration tests and all of them has been passed. 

Thanks for your attention.

**Comment by dakrone** [comment]
This has been resolved by #37401
